Implement autocorrection for Rails/ReadWriteAttribute cop.

huerlisi opened the pull request:
This implements autocorrection for the Rails/ReadWriteAttribute cop.

It is my first contribution to rubocop, therefore I'm not sure it fits your coding conventions;-)

jonas054 commented on lib/rubocop/cop/rails/read_write_attribute.rb:
```diff
@@ -18,6 +18,41 @@ module Rails
+          end
+        end
+
+        def autocorrect_read(node)
```
`autocorrect_read` and `autocorrect_write` should be private.

jonas054 commented on lib/rubocop/cop/rails/read_write_attribute.rb:
```diff
@@ -18,6 +18,41 @@ module Rails
+
+          selfing = "self[#{name_source}] = #{value_source}"
+
+          @corrections << lambda do |corrector|
```
Move these 3 duplicated lines into the `autocorrect` method and return the "selfing" value. After that change it would be better to name the methods, for example, `..._attribute_replacement`.

jonas054 commented on spec/rubocop/cop/rails/read_write_attribute_spec.rb:
```diff
@@ -16,4 +16,87 @@
   end
+
+  describe '#autocorrect' do
+    context 'write_attribute' do
```
What if there's an explicit receiver, `object.write_attribute(:attr, var)`? Leaving out the parentheses in one of the examples would also be good.

huerlisi replied:
Instances with explicit receivers are not marked as offenses, and therefore also not autocorrected.

I could change it to also record them as offenses, and implement the according autocorrection. Otherwise I'd add some spec documenting the current behaviour.

What do you think?

jonas054 replied:
Keep the behavior but add something to the spec that documents it, thanks!

huerlisi commented:
@jonas054 thank you for the code review, appreciated!

I've changed the code accordingly and rebased on current master. I've created independent commit for the issues you mentioned. Should I squash them again?

jonas054 commented:
The code looks good to me now apart from the missing explicit receiver in the specs that I mentioned, so please make that change and squash again. Maybe @bbatsov has some more comments.

huerlisi commented:
Squashed the autocorrection commits and added explicit specs.

bbatsov commented:
Doesn't look squashed to me.

huerlisi commented:
I've squashed the follow up fixes into the main autocorrection changes. I decidedly let it be 4 commits who each have some other context/target. Didn't want to have additional specs of current behavior and  new functionality in the same commit;-)

But I can squash it all together, as you like?

bbatsov commented:
@huerlisi The first 3 commits seem like a single task to me - I'd normally have them as a single commit. I'm a believer that related changes should live in a single commit.

huerlisi commented:
Sure, makes sense. Squashed as requested.

bbatsov commented:
Thanks! :+1:
